### ISSUE TITLE
dynamic Literal constructors behave badly

### ISSUE THREAD

**Issue by liss-h** [TortugaAttack/rdf4cpp#55]
Many of the Literal constructors behave badly as they allow the user to create a literal without the type being registered
which leads to triggered asserts in comparisons, ebv and numeric ops.

How to reproduce (used branch feature/literal-ops-comparisons for this example, but can be triggered in develop with num ops):

1. never mention `rdf4cpp::rdf::datatypes::xsd::String` anywhere in your code (because this would trigger the template instantiation and register the type, thus preventing this bug).

2. 
```c++
Literal lit1 = Literal{"test1"};
Literal lit2 = Literal{"test2"};
bool res = lit1 < lit2; // assertion triggered here
```

Constructors that have this behaviour:
- `Literal::Literal(std::string_view lexical_form, Node::NodeStorage &node_storage)`
- `Literal::Literal(std::string_view lexical_form, const IRI &datatype, Node::NodeStorage &node_storage)`
- `Literal::Literal(std::string_view lexical_form, std::string_view lang, Node::NodeStorage &node_storage)`
- `Literal Literal::make(std::string_view lexical_form, const IRI &datatype, Node::NodeStorage &node_storage)`

**Comment by liss-h** [comment]
Ideas:
- explicit template instantiations in cpp file
- static variables

**Comment by liss-h** [comment]
Fixed by #59
